Installation issue: julia

### Steps to reproduce the issue

```console
$ spack spec -I julia
 -   julia@1.11.8+openlibm+precompile build_system=makefile cpu_target=auto platform=linux os=almalinux9 target=zen4 %c,cxx,fortran=gcc@13.4.0
[+]      ^compiler-wrapper@1.0 build_system=generic platform=linux os=almalinux9 target=zen4 
[+]      ^curl@8.15.0~gssapi~ldap~libidn2~librtmp~libssh+libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=mbedtls,openssl platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[e]          ^gcc@13.4.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=almalinux9 target=x86_64 
[+]          ^gcc-runtime@13.4.0 build_system=generic platform=linux os=almalinux9 target=zen4 
[+]          ^openssl@3.6.0~docs+shared build_system=generic certs=mozilla platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^ca-certificates-mozilla@2025-08-12 build_system=generic platform=linux os=almalinux9 target=zen4 
[+]          ^pkgconf@2.5.1 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^dsfmt@2.2.5 build_system=makefile patches:=b79624c platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[e]      ^gcc@13.4.0+binutils+bootstrap~graphite+libsanitizer~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=almalinux9 target=x86_64 
[+]      ^gcc-runtime@13.4.0 build_system=generic platform=linux os=almalinux9 target=zen4 
[e]      ^glibc@2.34 build_system=autotools platform=linux os=almalinux9 target=x86_64 
[+]      ^gmake@4.4.1~guile build_system=generic platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^gmp@6.3.0+cxx build_system=autotools libs:=shared,static platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^autoconf@2.72 build_system=autotools platform=linux os=almalinux9 target=zen4 
[+]          ^automake@1.16.5 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^libtool@2.4.7 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]              ^findutils@4.10.0 build_system=autotools patches:=440b954 platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^m4@1.4.20+sigsegv build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^libsigsegv@2.14 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^libblastrampoline@5.15.0 build_system=makefile platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^libgit2@1.7.2~curl~ipo+mmap+ssh build_system=cmake build_type=Release generator=make https=system platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^cmake@3.31.9~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^pcre@8.45~jit+multibyte+pic+shared+static+utf build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^libssh2@1.11.1+shared build_system=autotools crypto=mbedtls platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^xz@5.6.3~pic build_system=autotools libs:=shared,static platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^libunwind@1.8.3~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components:=none libs:=shared,static patches:=4562c23 platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^libuv-julia@1.48.0 build_system=autotools commit=ca3a5a431a1c37859b6508e6b2a288092337029a platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^libwhich@1.1.0 build_system=generic platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^llvm@16.0.6+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug+link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~utils~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=none openmp=runtime patches:=4997cd3,b540ef6 shlib_symbol_version=JL_LLVM_16.0 targets:=all,amdgpu,bpf,nvptx,webassembly version_suffix=jl platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^binutils@2.45.1~debuginfod~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs:=shared,static platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^hwloc@2.12.2~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs:=shared,static platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^libpciaccess@0.17 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]                  ^util-macros@1.20.1 build_system=autotools platform=linux os=almalinux9 target=zen4 
[+]          ^libedit@3.1-20240808 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^libffi@3.5.2 build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^libxml2@2.13.5~http+pic~python+shared build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^lua@5.3.6+shared build_system=makefile fetcher=curl platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]              ^unzip@6.0 build_system=makefile patches:=179330d,24582ff,251d575,3371314,44599c8,47e9def,4e5a081,59c0983,64f6498,74bc961,7d8e5c7,81ca46c,881d2ed,aced0f2,b6f64d7,b7a14c3,c9a863e,ee9e260,f6f6236,f88b9d4,fde8f9d platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^ncurses@6.5-20250705~symlinks+termlib abi=none build_system=autotools patches:=7a351bc platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^ninja@1.13.0+re2c build_system=generic platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^re2c@3.1 build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^perl-data-dumper@2.173 build_system=perl platform=linux os=almalinux9 target=zen4 
[+]          ^swig@4.1.1 build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^mbedtls@2.28.9+pic build_system=makefile build_type=Release libs:=shared,static platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^mpfr@4.2.1 build_system=autotools libs:=shared,static patches:=3ec29a6 platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^autoconf-archive@2023.02.20 build_system=autotools platform=linux os=almalinux9 target=zen4 
[+]          ^texinfo@7.2~xs build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^nghttp2@1.59.0 build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^diffutils@3.12 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^openblas@0.3.30~bignuma~consistent_fpcsr+dynamic_dispatch+fortran+ilp64+locking+pic+shared build_system=makefile symbol_suffix=64_ threads=none platform=linux os=almalinux9 target=zen4 %c,cxx,fortran=gcc@13.4.0
[+]      ^openlibm@0.8.7 build_system=makefile platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^p7zip@17.05 build_system=makefile platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^patchelf@0.17.2 build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^pcre2@10.44~jit+multibyte+pic build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^perl@5.42.0+cpanm+opcode+open+shared+threads build_system=generic platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^bzip2@1.0.8~debug~pic+shared build_system=generic platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^gdbm@1.25 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^python@3.14.2+bz2+ctypes+dbm~debug~freethreading+libxml2+lzma~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib+zstd build_system=generic platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]          ^expat@2.7.3+libbsd build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^libbsd@0.12.2 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]                  ^libmd@1.1.0 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^gettext@0.23.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]              ^tar@1.35 build_system=autotools zip=pigz platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]                  ^pigz@2.8 build_system=makefile platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^readline@8.3 build_system=autotools patches:=21f0a03 platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^sqlite@3.50.4+column_metadata+fts+rtree build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^util-linux-uuid@2.41 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]          ^zstd@1.5.7+programs build_system=makefile compression:=none libs:=shared,static platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^suite-sparse@7.7.0~cuda~graphblas~openmp+pic build_system=generic platform=linux os=almalinux9 target=zen4 %c,cxx,fortran=gcc@13.4.0
[+]          ^metis@5.1.0~gdb~int64~ipo~no_warning~real64+shared build_system=cmake build_type=Release generator=make patches:=4991da9,93a7903,b1225da platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
[+]      ^utf8proc@2.10.0~ipo~shared build_system=cmake build_type=Release generator=make platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^which@2.21 build_system=autotools platform=linux os=almalinux9 target=zen4 %c=gcc@13.4.0
[+]      ^zlib-ng@2.3.2+compat+new_strategies+opt+pic+shared build_system=autotools platform=linux os=almalinux9 target=zen4 %c,cxx=gcc@13.4.0
```


### Error message

<details><summary>Error message</summary>

<pre>
     369    /usr/bin/ld: anonymous version tag cannot be combined with other version tags
  >> 370    collect2: error: ld returned 1 exit status
  >> 371    make[2]: *** [Makefile:109: /tmp/hpcxko3d/spack-stage/spack-stage-julia-1.11.8-4forpvn3u3o7fiddd7unqeohygsa5vku/spack-src/src/flisp/flisp] Error 1
     372    make[2]: Leaving directory '/tmp/hpcxko3d/spack-stage/spack-stage-julia-1.11.8-4forpvn3u3o7fiddd7unqeohygsa5vku/spack-src/src/flisp'
  >> 373    make[1]: *** [Makefile:368: flisp/libflisp.a] Error 2
     374    make[1]: Leaving directory '/tmp/hpcxko3d/spack-stage/spack-stage-julia-1.11.8-4forpvn3u3o7fiddd7unqeohygsa5vku/spack-src/src'
  >> 375    make: *** [Makefile:62: julia_flisp.boot.inc.phony] Error 2

</pre></details>
<details><summary>Additional information</summary>
The error occurs because `@JULIA_SHLIB_SYMBOL_VERSION@` is set in the expmap, but `@LLVM_SHLIB_SYMBOL_VERSION@` is not:

```
[hpcxko3d@login004 src]$ pwd
/tmp/hpcxko3d/spack-stage/spack-stage-julia-1.11.8-4forpvn3u3o7fiddd7unqeohygsa5vku/spack-src/src
[hpcxko3d@login004 src]$ diff julia.expmap.in julia.expmap
1c1
< @JULIA_SHLIB_SYMBOL_VERSION@ {
---
> JL_LIBJULIA_1.11 {
45c45
< @LLVM_SHLIB_SYMBOL_VERSION@ {
---
>  {
```

</pre></details>

### Information on your system

* **Spack:** 1.1.1
* **Builtin repo:** https://github.com/spack/spack-packages/commit/17ea6a6d483fc0bde1b2cca96ddaad0a10f2ef69
* **Python:** 3.9.25
* **Platform:** linux-almalinux9-zen4

### Additional information

[spack-build-out.txt](https://github.com/user-attachments/files/25095158/spack-build-out.txt)

[spack-build-env.txt](https://github.com/user-attachments/files/25095160/spack-build-env.txt)

@giordano  @haampie  @vchuravy

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/julia/package.py
+++ b/repos/spack_repo/builtin/packages/julia/package.py
@@ -223,6 +223,7 @@ class Julia(MakefilePackage):
     depends_on("libwhich", type="build")
     depends_on("which", type="build")  # for detecting 7z, lld, dsymutil
     depends_on("python", type="build")
+    depends_on("binutils", type="build")  # for readelf
 
     depends_on("blas")  # note: for now openblas is fixed...
     depends_on("curl tls=mbedtls +nghttp2 +libssh2")

